app/controllers/sessions_controller.rb

```diff
@@ -8,28 +8,46 @@ class SessionsController < ApplicationController
   include RateLimiting
 
   skip_before_filter :authenticate_user!, :check_if_password_expired,
-                     only: [:new, :lookup_login, :authenticate,
+                     only: [:start, :lookup_login, :authenticate,
                             :create, :failure, :destroy, :email_usernames]
 
   skip_before_filter :complete_signup_profile, only: [:destroy]
 
-  before_filter :save_new_params_in_session, only: [:new]
-  before_filter :store_authorization_url_as_fallback, only: [:new, :create]
-  before_filter :maybe_skip_to_sign_up, only: [:new]
+  before_filter :save_new_params_in_session, only: [:start]
+  before_filter :store_authorization_url_as_fallback, only: [:start, :create]
+
+  before_filter :maybe_skip_to_sign_up, only: [:start]
 
   before_filter :allow_iframe_access, only: :reauthenticate
 
 
-  # If the user arrives to :new already logged in, this means they got linked to
+  # If the user arrives to :start already logged in, this means they got linked to
   # the login page somehow; attempt to redirect to the authorization url stored
   # earlier
-  before_filter :redirect_back, if: -> { signed_in? }, only: :new
+  before_filter :redirect_back, if: -> { signed_in? }, only: :start
 
   fine_print_skip :general_terms_of_use, :privacy_policy,
-                  only: [:new, :lookup_login, :authenticate, :create, :failure, :destroy, :redirect_back, :email_usernames]
+                  only: [:start, :lookup_login, :authenticate, :create, :failure, :destroy, :redirect_back, :email_usernames]
 
   # Login form
-  def new; end
+  def start
+    handle_with(SessionsStart,
+                signup_state: signup_state,
+                session_management: self,
+                success: lambda do
+                  if @handler_result.outputs.user
+                    redirect_back
```

Comment:
I think this needs to be `redirect_to action: :redirect_back` so that the `before_filter` that checks for terms signature is run.  Can you add a feature spec where a LMS user logs in but the terms have changed (so that they have unsigned terms) and the spec shows that they have to sign terms before getting sent back to the app?  See `make_new_contract_version` in `spec/feature_helpers.rb`.  I think the general case spec is missing for this too, which is my bad.

Reply:
I added the spec as you suggested (good call, doing so uncovered a different bug).

However it still didn't prompt for terms - when I went to track down why I discovered that the `redirect_back` action is set to skip fine_print https://github.com/openstax/accounts/blob/master/app/controllers/sessions_controller.rb#L29

That's contrary to the comment before the action though.  Is the `fine_print_skip` in error?